### PR TITLE
Functional endpoints

### PR DESCRIPTION
**Pull request by david-smith-ft**

### CODE REVIEW

File: smartlogic/handlers.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	rw.Header().Set("X-Request-Id", tid)
+
+	var smLogicConcept = SmartlogicConcept{}
+	decoder := json.NewDecoder(req.Body)
+
+	err := decoder.Decode(&smLogicConcept)
+
+	if err != nil {
+		log.Errorf("Error %v whilst processing json body", err)
+		rw.WriteHeader(http.StatusServiceUnavailable)
```
Comment (scott-ace-newton) [inline smartlogic/handlers.go]: Would this not be a 400 instead of 503?

File: smartlogic/handlers.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	tid := transactionidutils.GetTransactionIDFromRequest(req)
+	rw.Header().Set("Content-Type", "application/json")
+	rw.Header().Set("X-Request-Id", tid)
+
+	var smLogicConcept = SmartlogicConcept{}
+	decoder := json.NewDecoder(req.Body)
+	err := decoder.Decode(&smLogicConcept)
+	if err != nil {
+		log.Errorf("Error %v whilst processing json body", err)
+		rw.WriteHeader(http.StatusServiceUnavailable)
```
Comment (scott-ace-newton) [inline smartlogic/handlers.go]: Same as above

File: smartlogic/handlers.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+func (h *SmartlogicConceptTransformerHandler) gtgCheck(rw http.ResponseWriter, req *http.Request) {
+	if errString, err := h.kafkaHealthCheck().Checker(); err != nil {
+		log.WithError(err).Errorf("Kafka healthcheck failed [%s]", errString)
+		rw.WriteHeader(http.StatusServiceUnavailable)
+		rw.Write([]byte("Kafka healthcheck failed"))
+		return
+	}
+	if errString, err := h.conceptRwHealthCheck().Checker(); err != nil {
+		log.WithError(err).Errorf("Concept RW S3 healthcheck failed [%s]: %s", errString)
```
Comment (david-smith-ft) [inline smartlogic/handlers.go]: `missing argument for Errorf("%s"): format reads arg 2, have only 1 args`

**Comment by david-smith-ft** [comment]
Approved (if I could)

**Pull request merged**